channels/audin/client/audin_main.c

```diff
@@ -416,22 +416,24 @@ static BOOL audin_open_device(AUDIN_PLUGIN* audin, AUDIN_CHANNEL_CALLBACK* callb
 	if (!supported)
 	{
 		/* Default sample rates supported by most backends. */
-		const UINT32 samplerates[] = { 96000, 48000, 44100, 22050 };
+		const UINT32 samplerates[] = { format.nSamplesPerSec, 96000, 48000, 44100, 22050 };
 		BOOL test = FALSE;
 
 		format.wFormatTag = WAVE_FORMAT_PCM;
 		format.wBitsPerSample = 16;
-		test = IFCALLRESULT(FALSE, audin->device->FormatSupported, audin->device, &format);
-		if (!test)
+		format.cbSize = 0;
+		size_t x;
+		for (x = 0; x < ARRAYSIZE(samplerates); x++)
 		{
-			size_t x;
-			for (x = 0; x < ARRAYSIZE(samplerates); x++)
-			{
-				format.nSamplesPerSec = samplerates[x];
-				test = IFCALLRESULT(FALSE, audin->device->FormatSupported, audin->device, &format);
-				if (test)
-					break;
-			}
+			format.nSamplesPerSec = samplerates[x];
+			format.nChannels = audin->format->nChannels;
+			test = IFCALLRESULT(FALSE, audin->device->FormatSupported, audin->device, &format);
+			if (test)
+				break;
+			format.nChannels = 3 - format.nChannels;
```

Comment:
looks strange, so you want to check if the format is compatible with different channel count? better run a loop instead of magic (e.g. `3 - 1` or `3 - 2` in this case here)

Reply:
I have found my method more elegant than a loop because I did not want to simply iterate over 1 and 2 channels. I want that iteration always begins with the same number of channels as the currently chosen format.
I don't think that using a loop with that kind of behavior will be simpler. Actually, I don't see how to express that particular behavior without doing something "strange" or excessively verbose.

Reply:
well, `3 ` is a magic number here, that is why I did ask. Not that important though.